### PR TITLE
Implement platform logging for App Check

### PR DESCRIPTION
**Pull request by rosalyntan**

### CODE REVIEW

File: appcheck/firebase-appcheck/src/main/java/com/google/firebase/appcheck/internal/NetworkClient.java
Comment (vkryachko) [inline appcheck/firebase-appcheck/src/main/java/com/google/firebase/appcheck/internal/NetworkClient.java:71]: Consider go/di-practices/inject-direct-dependencies

Comment (rosalyntan) [reply appcheck/firebase-appcheck/src/main/java/com/google/firebase/appcheck/internal/NetworkClient.java:71]: Thanks for the link! I passed `FirebaseApp` directly into `NetworkClient` to avoid adding the `:appcheck:firebase-appcheck-interop` dependency to `firebase-appcheck-safetynet.gradle` and `firebase-appcheck-debug.gradle` (for accessing `FirebaseAppCheck#getInstance`).

I think it's fine to add the interop dependency, but it seems cleaner to only have that dependency in the core App Check SDK.

Comment (vkryachko) [reply appcheck/firebase-appcheck/src/main/java/com/google/firebase/appcheck/internal/NetworkClient.java:71]: I see, one thing I would suggest is to consider injecting the NetworkClient via firebase components instead of having both `safetynet` and `debug` sdks creating instantiating it on their own. This would also allow you to register both of these sdks with the platform logging user-agent.

(Optional)

**Review by vkryachko — COMMENTED** [review]
Overall LGTM,

Please get a review by @VinayGuthal before merging

**Review by VinayGuthal — APPROVED** [review]

File: appcheck/firebase-appcheck/src/main/java/com/google/firebase/appcheck/internal/NetworkClient.java
Comment (rosalyntan) [reply appcheck/firebase-appcheck/src/main/java/com/google/firebase/appcheck/internal/NetworkClient.java:71]: Ack, that makes sense! I've filed b/186549222 to follow up on this in a future PR.

**Pull request merged**
